From 7400c9f9f3dded17af17533c6c8545d0c51f9993 Mon Sep 17 00:00:00 2001
From: Zeeshan Ali (Khattak) <zeeshanak@gnome.org>
Date: Mon, 17 May 2010 20:06:18 +0300
Subject: [PATCH 2/2] Handle multiple variables inside property node

Although we stopped doing this for very good reasons in commit 7ebd4453,
we are now re-introducing this to be compatible with server-side
implementations using older GUPnP.
---
 libgupnp/gupnp-service-proxy.c |   17 +++++++++++------
 1 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/libgupnp/gupnp-service-proxy.c b/libgupnp/gupnp-service-proxy.c
index 9db8634..33e06f0 100644
--- a/libgupnp/gupnp-service-proxy.c
+++ b/libgupnp/gupnp-service-proxy.c
@@ -1630,12 +1630,17 @@ emit_notifications_for_doc (GUPnPServiceProxy *proxy,
         for (node = node->children; node; node = node->next) {
                 xmlNode *var_node;
 
-                /* variableName node */
-                var_node = node->children;
-
-                if (var_node != NULL &&
-                    strcmp ((char *) node->name, "property") == 0)
-                        emit_notification (proxy, var_node);
+                /* Although according to the UPnP specs, there should be only
+                 * one variable node inside a 'property' node, we still need to
+                 * entertain the possibility of multiple variables inside it to
+                 * be compatible with implementations using older GUPnP.
+                 */
+                for (var_node = node->children;
+                     var_node;
+                     var_node = var_node->next) {
+                        if (strcmp ((char *) node->name, "property") == 0)
+                                emit_notification (proxy, var_node);
+                }
         }
 }
 
-- 
1.7.0.4

